querydsl-jpa/src/test/java/com/querydsl/jpa/HibernateHandlerTest.java

```diff
@@ -0,0 +1,115 @@
+package com.querydsl.jpa;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javax.persistence.Query;
+
+import org.hibernate.query.NativeQuery;
+import org.hibernate.query.spi.ScrollableResultsImplementor;
+import org.hibernate.transform.ResultTransformer;
+import org.junit.Test;
+
+import com.mysema.commons.lang.IteratorAdapter;
+import com.querydsl.core.types.FactoryExpression;
+import com.querydsl.jpa.domain4.Library;
+
+import static org.easymock.EasyMock.anyObject;
+import static org.easymock.EasyMock.createMock;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
+import static org.easymock.EasyMock.verify;
+import static org.hibernate.ScrollMode.FORWARD_ONLY;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class HibernateHandlerTest {
```

Comment:
Cool. Could you please add tests that assert the exception is thrown from these methods? When you do, it'd be good if the test names would be more indicative of the given state. e.g addEntity_validNativeQuery_shouldAddScalar.

Reply:
sure. I will do. :)

Reply:
Thanks. Also, please note @jwgmeligmeyling review. :)

Reply:
I added thrown test case and pushed new commit 'refactor: added thrown test case for HibernateHandlerTest and removed conditional statement'